QVM weblet

0oM4R opened the pull request:
### Description
As in threefoldtech/tfgrid-sdk-ts#126, VM with QSFS solution added to weblet

### Changes

- Add QVM  to 

> 1. App.vue
> 2. editor.vue
> 3. App.svelte
> 4. DeployedList.wc.svelte
> 5. currentDeployment.ts _with QSFS_
> 6. solutionsList.ts

- Create QSFS type 
- Create Qvm element
- Create DeployQvm
- Add QSFS to DeployVM.ts
- Add qsfsSisk to Vm type
- Add some validation in validateName.ts
- Add delete qsfs in deleteDeployment.ts
- Add metrics label to Table.svelte to use it in view qsfs metrics 
- use Multiselect element in selectNodeId.svelte
- modify MultiSelect.svelte layout
### Related Issues

List of related issues
threefoldtech/tfgrid-sdk-ts#126 
### Checklist


- [x] Build pass
- [x] Code format and docstrings

0oM4R committed:
add front end for qvm

0oM4R committed:
update ui

0oM4R committed:
frontend in progress

maayarosama committed:
updates

0oM4R committed:
rename qsfs proberties

0oM4R committed:
add multiselect option:

MohamedElmdary committed:
support multi nodeIds in SelectNodeId

0oM4R committed:
update qsfs type

0oM4R committed:
Merge branch 'development' of github.com:threefoldtech/grid_weblets into development_qvm_weblet

0oM4R committed:
add mulit select component

0oM4R committed:
cleanup

0oM4R committed:
add qsfs type to vm

0oM4R committed:
refactor some lines, mount point in progress

0oM4R committed:
refactor qsfs type

0oM4R committed:
deployQvm
in progress add secret validation, and add hru to selectNodeId

0oM4R committed:
Merge branch 'development' of github.com:threefoldtech/grid_weblets into development_qvm_weblet

0oM4R committed:
Qvm in progress
add some validation
add Qvm type
handle Qvm in deployVm

0oM4R committed:
in progress

0oM4R committed:
add with deploy

0oM4R committed:
add some console log to debug

0oM4R committed:
idk why its work

0oM4R committed:
now i know

0oM4R committed:
Merge branch 'development' of github.com:threefoldtech/grid_weblets into development_qvm_weblet

0oM4R committed:
apply multiSelect comp in qvm

0oM4R committed:
cleanup selectNodeId

0oM4R committed:
modify multiselect style

0oM4R committed:
edit qsfs

0oM4R committed:
remover log

0oM4R committed:
solve ux issues

0oM4R committed:
add qvm to deployed list

0oM4R committed:
cleanup selectNodeID

0oM4R committed:
add qvm to solutionList

0oM4R committed:
add qsfs to deployVm

0oM4R committed:
format file

0oM4R committed:
format file

0oM4R committed:
config metadata

0oM4R committed:
remove unnecessary include

0oM4R committed:
remove svelte-multiselect package

0oM4R committed:
add qvm

0oM4R committed:
remove active status

0oM4R committed:
handle delete qsfs on delete vm

0oM4R committed:
apply some issue requirements

0oM4R committed:
add metrics button to deployment list

0oM4R committed:
Merge branch 'development' of github.com:threefoldtech/grid_weblets into HEAD

0oM4R committed:
cleanup

0oM4R committed:
cleanup

0oM4R committed:
cleanup

xmonader reviewed (changes requested):
lgtm

AhmedHanafy725 commented on src/utils/deployQVM.ts:
```diff
@@ -0,0 +1,41 @@
+  const metadate = {
+    type: "QSFS",
+    name: name,
+    projectName: `QSFS of vm ${vm.name}`,
```
that's too long for the project name, can't it be `QVM`?

AhmedHanafy725 commented on src/utils/validateName.ts:
```diff
@@ -269,3 +269,14 @@ export function validateAlgoStorage(value: string, net, type): string | void {
 }
+
+export function validateQsfsSecret(value: string): string | void{
+  if (!value.length) return `zDB password is required`
```
```suggestion
  if (!value.length) return `ZDB password is required`
```

AhmedHanafy725 commented on src/utils/validateName.ts:
```diff
@@ -269,3 +269,14 @@ export function validateAlgoStorage(value: string, net, type): string | void {
+  if (!NUM_REGEX.test(value.toString()) || isNaN(+value))
+  return "zDB count must be a valid number.";
+  else if(value < 3) return `Minimum allowed zDB is 3 zDBs`;
+  
```
```suggestion
  return "ZDB count must be a valid number.";
  else if(value < 3) return `Minimum allowed ZDB is 3 ZDBs`;
  
```

AhmedHanafy725 commented on src/elements/DeployedList/DeployedList.wc.svelte:
```diff
@@ -456,6 +458,23 @@
+          type: "warning",
+          label: "Metrics",
+          click: (_, i) => {
+            const domain = rows[i].details.mounts[0]["metricsEndpoint"];
```
this endpoint is only available on the planetary network, should it be exposed over the gateway? @xmonader

AhmedHanafy725 replied:
let's remove this button for the metrics endpoint, and just keep it in the deployment details

AhmedHanafy725 commented on src/elements/qvm/Qvm.wc.svelte:
```diff
@@ -0,0 +1,340 @@
+    {
+      label: "Disk size (MB) ",
+      symbol: "disk",
+      placeholder: "Memory of each ZDB in MB",
```
this is confusing, is it disk or memory?

0oM4R replied:
resolved to be "Disk size"

0oM4R committed:
Update src/utils/validateName.ts

Co-authored-by: AhmedHanafy725 <41957921+AhmedHanafy725@users.noreply.github.com>

0oM4R committed:
Update src/utils/validateName.ts

Co-authored-by: AhmedHanafy725 <41957921+AhmedHanafy725@users.noreply.github.com>

0oM4R committed:
Update deployQVM.ts

0oM4R committed:
resolve disk field

https://github.com/threefoldtech/grid_weblets/pull/1102#discussion_r1022072603

0oM4R committed:
remove this button for the metrics endpoint

AhmedHanafy725 commented on src/elements/qvm/Qvm.wc.svelte:
```diff
@@ -0,0 +1,340 @@
+      invalid: false,
+    },
+    {
+      label: "Disk size (MB) ",
+      symbol: "disk",
+      placeholder: "Disk size of each ZDB in MB",
```
disk size in all weblets is in GB, can you convert it to be consistent?

0oM4R committed:
change MB to GB in Qvm disk size

0oM4R committed:
solve bug and enhance /
update options array to be empty if there is no nodes available , add indicator how many nodes founded

0oM4R committed:
enhance UXupdate nodes if count or number of nodes changed
